flax/experimental/nnx/nnx/rnglib.py

```diff
@@ -173,7 +160,7 @@ def fork(
   state: State,
   split_filter: filterlib.Filter,
   split_pattern: SplitPattern,
-) -> tuple[State, State]:
+) -> tuple[State, State, State, State]:
```

Comment:
I would recommend making this a `typing.NamedTuple` or a dataclass so that the caller doesn't have to remember what each `State` component corresponds to.

Reply:
done